sys/net/application_layer/nanocoap/sock.c

```diff
@@ -267,7 +280,6 @@ ssize_t nanocoap_sock_request_cb(nanocoap_sock_t *sock, coap_pkt_t *pkt,
                 DEBUG("nanocoap: timeout waiting for response\n");
                 timeout *= 2;
                 deadline = _deadline_from_interval(timeout);
-                state = STATE_REQUEST_SEND;
                 continue;
```

Comment:
We are now stuck in this loop if there really is no response from the server. (e.g. we send a request to a non-existing address)